modules/fargate/fargate/main.tf

```diff
@@ -0,0 +1,360 @@
+  }
+
+  # NOTE: TF is unable to destroy a target group while a listener is attached,
+  # therefor we have to create a new one before destroying the old. This also means
+  # we have to let it have a random name, and then tag it with the desired name.
+  lifecycle {
+    create_before_destroy = true
+  }
+  tags = merge(
+    var.tags,
+    lookup(var.containers_definitions[each.key], "task_tags", {}),
+    {
+      Name = "${each.key}-target-lookup-${lookup(var.containers_definitions[each.key], "task_container_port", "")}"
+    },
+  )
+}
+
+# ------------------------------------------------------------------------------
+# ECS Task/Service
+# ------------------------------------------------------------------------------
+resource "aws_ecs_task_definition" "task" {
+  for_each = var.containers_definitions
+
+  depends_on               = [null_resource.lb_exists]
+  family                   = lookup(var.containers_definitions[each.key], "task_container_name", each.key)
+  execution_role_arn       = aws_iam_role.execution[each.key].arn
+  network_mode             = "awsvpc"
+  requires_compatibilities = ["FARGATE"]
+  cpu                      = lookup(var.containers_definitions[each.key], "task_definition_cpu", 256)
+  memory                   = lookup(var.containers_definitions[each.key], "task_definition_memory", 1024)
+  task_role_arn            = aws_iam_role.task[each.key].arn
+
+  container_definitions = <<EOF
+[{
+    "name": "${lookup(var.containers_definitions[each.key], "task_container_name", each.key)}",
+    "image": "${lookup(var.containers_definitions[each.key], "task_container_image", "nginx")}",
+    %{if lookup(var.containers_definitions[each.key], "task_repository_credentials", "") != ""~}
+    "repositoryCredentials": {
+        "credentialsParameter": "${lookup(var.containers_definitions[each.key], "task_repository_credentials", null)}"
+    },
+    %{~endif}
+    "essential": true,
+    "portMappings": [
+        {
+            "containerPort": ${lookup(var.containers_definitions[each.key], "task_container_port", null)},
+            "hostPort": ${lookup(var.containers_definitions[each.key], "task_container_port", null)},
+            "protocol":"tcp"
+        }
+    ],
+    "logConfiguration": {
+        "logDriver": "awslogs",
+        "options": {
+            "awslogs-group": "${each.key}",
+            "awslogs-region": "${data.aws_region.current.name}",
+            "awslogs-stream-prefix": "container"
+        }
+    },
+    "command": ${jsonencode(lookup(var.containers_definitions[each.key], "task_container_command", []))},
+    "environment": ${jsonencode(lookup(var.containers_definitions[each.key], "task_container_environment", []))}
+}]
+EOF
+  tags = merge(
+    var.tags,
+    lookup(var.containers_definitions[each.key], "task_tags", {})
+  )
+}
+
+resource "aws_ecs_service" "service_with_no_service_registries" {
```

Comment:
Is it not possible to use a dynamic block for `service_registries` and avoid the conditional here?